// SPDX-FileCopyrightText: 2025 Carnegie Mellon University - Living Edge Lab
//
// SPDX-License-Identifier: GPL-2.0-only

syntax = "proto3";
package protocol.control;

import "common.proto";
import "google/protobuf/any.proto";

/*
 * Directive for remote control from the Swarm Controller
 */
message ControlRequest {
  protocol.common.Request request = 1;
  google.protobuf.Any control_message = 2;
}

message ControlResponse {
  protocol.common.Response response = 1;
  google.protobuf.Any control_result = 2;
}

/*
 * Mode switch for control over the vehicle
 * REMOTE -> Swarm Controller
 * LOCAL -> Mission
 * RC -> Remote control joystick
 */
enum ControlMode {
  REMOTE = 0;
  LOCAL = 1;
  RC = 2;
}
